src/Interpreters/InterpreterSelectQuery.cpp

```diff
@@ -831,6 +833,33 @@ static SortDescription getSortDescription(const ASTSelectQuery & query, ContextP
     return order_descr;
 }
 
+static InterpolateDescription getInterpolateDescription(const ASTSelectQuery & query, Block block, ContextPtr context)
+{
+    InterpolateDescription interpolate_descr;
+    if (query.interpolate())
+    {
+        interpolate_descr.reserve(query.interpolate()->children.size());
+
+        for (const auto & elem : query.interpolate()->children)
+        {
+            auto interpolate = elem->as<ASTInterpolateElement &>();
+            ColumnWithTypeAndName *block_column = block.findByName(interpolate.column->getColumnName());
+            if (!block_column)
+                throw Exception(ErrorCodes::UNKNOWN_IDENTIFIER,
+                    "Missing column '{}' as an INTERPOLATE expression target", interpolate.column->getColumnName());
+            ColumnWithTypeAndName column = block_column->cloneEmpty();
+
+            auto syntax_result = TreeRewriter(context).analyze(interpolate.expr, block.getNamesAndTypesList());
+            ExpressionAnalyzer analyzer(interpolate.expr, syntax_result, context);
+            ExpressionActionsPtr actions = analyzer.getActions(true, true, CompileExpressions::yes);
```

Comment:
If we call `getActionsDAG` here, we can get DAG which could be modified.
Here we can also check if the result type is the same as initial, and add `CAST` function. There are possible options:
* use `ActionsDAG::makeConvertingActions`, and then merge two DAGs with `ActionsDAG::merge`
* iterate `ActionsDAG::getIndex()` and add function CAST manually, something like [here](https://github.com/ClickHouse/clickhouse/blob/5ef83deaa68a6d4c3f6636a6fabe52cfa332c088/src/Interpreters/ActionsDAG.cpp#L1087-L1103)